core/modules/dashboard/includes/block.news.inc

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * @file
+ * Dashboard block displaying BackdropCMS.org news items.
+ */
+
+class DashboardNewsBlock extends Block {
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($plugin_name, array $data) {
+    parent::__construct($plugin_name, $data);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTitle() {
+    return !empty($this->settings['title']) ? check_plain($this->settings['title']) : t('Backdrop News');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAdminTitle() {
+    return t('Backdrop News');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAdminPreview() {
+    return t('Displays a feed of news from BackdropCMS.org.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getContent() {
+    $feed = backdrop_http_request('https://backdropcms.org/api/backdrop-news');
+    $news = json_decode($feed->data);
+    $links = array();
+
+    if ($feed->code === '200') {
+      foreach ($news->nodes as $key => $node) {
+        $links[] = l(
+          $node->node->title,
+          $node->node->field_link
+        ) . ' - ' . $node->node->field_tagline;
+      }
+    }
```

Comment:
Would it be worth checking if the error is 404 etc and changing the message to coulndt connect or offline or something?

Reply:
that happens in the `if (empty($links))`

Reply:
I mean change the message. If 200 and empty, `'#markup' => t('No news at this time.')`, if 404 `'#markup' => t('Could not connect to server.')` etc. Not sure its worth all that though.